server/worker/apple_mdm.go

```diff
@@ -204,6 +220,30 @@ func (a *AppleMDM) runPostDEPEnrollment(ctx context.Context, args appleMDMArgs)
 	return nil
 }
 
+// getTeamConfig gets team config from DB if not provided.
+func (a *AppleMDM) getTeamConfig(ctx context.Context, team *fleet.Team, teamID uint) (*fleet.Team, error) {
+	if team == nil {
+		var err error
+		team, err = a.Datastore.Team(ctx, teamID)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "fetch team to send AccountConfiguration")
+		}
+	}
+	return team, nil
+}
+
+// getAppConfig gets app config from DB if not provided.
+func (a *AppleMDM) getAppConfig(ctx context.Context, appConfig *fleet.AppConfig) (*fleet.AppConfig, error) {
```

Comment:
Is the addition of the config parameter motivated by testing concerns or to optimize datastore calls? If the latter, doesn't the in-memory app config cache already address that concern (in which case, I wonder if we're approaching a point where we should consider a cache for team config as well).